src/dotnet/UI.Blazor/Services/EscapeHandler.cs

```diff
@@ -0,0 +1,38 @@
+using ActualChat.UI.Blazor.Module;
+
+namespace ActualChat.UI.Blazor.Services;
+
+public sealed class EscapeHandler : IAsyncDisposable
+{
+    private readonly DotNetObjectReference<EscapeHandler> _blazorRef;
+    private readonly IJSRuntime _jsRuntime;
+
+    private IJSObjectReference? _jsRef;
+
+    public event EventHandler? Escape;
+
+    public EscapeHandler(IJSRuntime jsRuntime)
+    {
+        _blazorRef = DotNetObjectReference.Create(this);
+        _jsRuntime = jsRuntime;
+    }
+
+    [JSInvokable]
+    public void OnEscape()
+        => Escape?.Invoke(this, EventArgs.Empty);
```

Comment:
One other downside of this design is that you can't handle escape purely in JS - e.g. you can't hide the content there on escape & don't even trigger any interop (remember it's a networking call in Blazor Server) when you need it. And it might be a frequent scenario - e.g. you can at least hide almost any element instantly first, and only after that trigger the Blazor call to do the rest (i.e. actually remove the content from DOM & dispose stuff).

So that's why I said it makes sense to implement a JS-based API  - I was assuming you start from it & add its Blazor counterpart after.